index.html

```diff
@@ -1976,41 +1988,66 @@ <h3>
           The steps for <dfn>processing the <code>start_url</code> member</dfn>
           are given by the following algorithm. The algorithm takes a
           <a>USVString</a> <var>value</var>, a <a>URL</a> <var>manifest
-          URL</var>, and a <a>URL</a> <var>document URL</var>. This algorithm
-          returns a <a>URL</a>.
+          URL</var>, and an optional <a>URL</a> <var>document URL</var>. This
+          algorithm returns a <a>URL</a>.
         </p>
         <ol>
+          <li>If <var>document URL</var> is given, and <var>manifest URL</var>
+          is not <a>same origin</a> as <var>document URL</var>, let
+          <var>default</var> be <var>document URL</var>.
+          </li>
+          <li>Otherwise, let <var>default</var> be the result of <a>parsing</a>
+          ".", using <var>manifest URL</var> as the <var>base</var> URL.
+          </li>
           <li>If <var>value</var> is the empty <a>string</a>, return
-          <var>document URL</var>.
+          <var>default</var>.
           </li>
           <li>Let <var>start URL</var> be the result of <a>parsing</a>
           <var>value</var>, using <var>manifest URL</var> as the
           <var>base</var> URL.
           </li>
           <li>If <var>start URL</var> is failure:
-            <ul>
+            <ol>
               <li>
                 <a>Issue a developer warning</a>.
               </li>
-              <li>Return <var>document URL</var>.
+              <li>Return <var>default</var>.
               </li>
-            </ul>
+            </ol>
           </li>
-          <li>If <var>start URL</var> is not <a>same origin</a> as
-          <var>document URL</var>:
+          <li>If <var>document URL</var> is given, and <var>start URL</var> is
+          not <a>same origin</a> as <var>document URL</var>:
```

Comment:
If the document URL is not given, and the start_url has a different origin from the manfiest_url, should we require/recommend the parser of the manifest to verify that the start_url's origin acknowledge this manifest_url?